pkg/services/vimmachine.go

```diff
@@ -41,18 +43,20 @@ import (
 )
 
 // VimMachineService reconciles VSphere VMs.
-type VimMachineService struct{}
+type VimMachineService struct {
+	Client client.Client
+}
 
 // FetchVSphereMachine returns a new MachineContext containing the vsphereMachine.
-func (v *VimMachineService) FetchVSphereMachine(c client.Client, name types.NamespacedName) (capvcontext.MachineContext, error) {
+func (v *VimMachineService) FetchVSphereMachine(name types.NamespacedName) (capvcontext.MachineContext, error) {
```

Comment:
I think we should also have `ctx context.Context` as parameter here and use it instead of `context.Background`. 

I think this is not on the list for context refactoring? We can also add it to the list to not increase scope of this PR.

Reply:
Good catch 👍